### PR TITLE
Refactor and add missing test case for TrailingComma mixin

### PR DESCRIPTION
**Pull request by Drenmi**
I was working on something else, and came across [this line of code](https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/mixin/trailing_comma.rb#L101) in `Mixin/TrailingComma`.

It didn't seem to be doing anything, and removing it still passed all the tests. However, the comment allowed me to reverse engineer the edge cases that this line covers, so I added a unit test for it.

After doing that, I poked around a bit in the method, and it started to untangle, resulting in a much simpler implementation that still passes all the tests.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/trailing_comma_in_arguments_spec.rb [hunk 1]
```diff
@@ -212,8 +212,8 @@
 
       it 'accepts an empty hash being passed as a method argument' do
         expect_no_offenses(<<-RUBY.strip_indent)
-          Foo.new([
-                   ])
+          Foo.new({
```
Comment (Drenmi) [inline spec/rubocop/cop/style/trailing_comma_in_arguments_spec.rb]: This assertion didn't match its description, so I changed it.

**Comment by bbatsov** [comment]
Awesome work!
